lib/chibi/math/prime.scm

```diff
@@ -175,57 +198,105 @@
          ((prime? n) n)
          (else (lp (+ n 2)))))))))
 
+;; Given an initial value \var{r1} representing the (empty)
+;; factorization of 1 and a procedure \var{put}
+;; (called as \scheme{(\var{put} \var{r} \var{p} \var{k})})
+;; that, given prior representation \var{r},
+;; adds a prime factor \var{p} of multiplicity \var{k},
+;; returns a factorization function which returns the factorization
+;; of its non-zero integer argument \var{n} in this representation.
+;; The optional 3rd and 4th arguments, if provided, specialize \var{put}
+;; for particular primes:
+;; \var{put2}  for \var{p}=2,  called as \scheme{(\var{put2} \var{r} \var{k})})
+;; \var{put-1} for \var{p}=-1, called as \scheme{(\var{put-1} \var{r})}).
+(define (make-factorizer r1 put . o)
+  (let-optionals o ((put2  (lambda (r k) (put r  2 k)))
+		    (put-1 (lambda (r)   (put r -1 1))))
```

Comment:
The indentation is off here and in other places, it looks like you're using tabs.

Apologies if this isn't your preferred style, but the Chibi codebase forbids tabs.
You can get a warning about this if you run `make init-dev`.  Please use spaces.

Reply:
Actually, I hate tabs too.  It's just that too many of my projects were with people that preferred them so it became my default at some point.  Emacs makes this stuff invisible.  But yay, I can fix.

- is there a `CONTRIBUTING` file that I missed (or some other convention I don't know about?); might be worth adding one.

- How do you feel about rebasing?  (I'd just as soon redo things so that it's tabless from the beginning, and also the git log on this pull branch is getting pretty tangled)

Reply:
I'll add a note to the README.  Rebasing is fine.

Reply:
Just so that you know, github apparently [gives brownie points](https://github.com/ashinn/chibi-scheme/community) if there's [a separate CONTRIBUTING file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors).